### PR TITLE
Add columns to breakpoints

### PR DESCRIPTION
**Pull request by annagrin**
This change allows setting breakpoints on dart lines that are compiled into the middle of longer JS lines. For example, in the following code `Scaffold(...)` is compiled into one JavaScript line, which did not let us set a breakpoint on `// Breakpoint on this line` line before:
 
```dart
Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar( // Breakpoint on this line
        title: Text(widget.title),
      ),
    ... 
   );
}
```

### CODE REVIEW

File: dwds/lib/src/debugging/debugger.dart [hunk 1]
```diff
@@ -723,11 +728,16 @@ bool isNativeJsObject(InstanceRef instanceRef) {
 
 /// Returns the Dart line number for the provided breakpoint.
 int _lineNumberFor(Breakpoint breakpoint) =>
-    int.parse(breakpoint.id.split('#').last);
+    int.parse(breakpoint.id.split('#').last.split(':').first);
+
+/// Returns the Dart line number for the provided breakpoint.
```
Comment (nshahan) [inline dwds/lib/src/debugging/debugger.dart:733]: line number -> column number

File: dwds/test/build_daemon_breakpoint_test.dart [hunk 2]
```diff
@@ -74,6 +74,26 @@ void main() {
...
+
+        await stream.firstWhere(
+            (Event event) => event.kind == EventKind.kPauseBreakpoint);
+
+        expect(bp, isNotNull);
+        expect(
+            bp.location,
+            isA<SourceLocation>()
+                .having((loc) => loc.script, 'script', equals(mainScript))
+                .having((loc) => loc.line, 'line', equals(line)));
```
Comment (nshahan) [inline dwds/test/build_daemon_breakpoint_test.dart:91]: Should we add an expectation for the column too?

File: dwds/test/frontend_server_breakpoint_test.dart [hunk 3]
```diff
@@ -104,6 +104,26 @@ void main() {
...
+
+        await stream.firstWhere(
+            (Event event) => event.kind == EventKind.kPauseBreakpoint);
+
+        expect(bp, isNotNull);
+        expect(
+            bp.location,
+            isA<SourceLocation>()
+                .having((loc) => loc.script, 'script', equals(mainScript))
+                .having((loc) => loc.line, 'line', equals(line)));
```
Comment (nshahan) [inline dwds/test/frontend_server_breakpoint_test.dart:122]: and here?

File: dwds/lib/src/debugging/debugger.dart [hunk 4]
```diff
@@ -324,10 +328,11 @@ class Debugger extends Domain {
     var frame = e.params['callFrames'][0];
     var location = frame['location'];
     var scriptId = location['scriptId'] as String;
-    var lineNumber = location['lineNumber'] as int;
+    var line = location['lineNumber'] as int;
+    var column = location['columnNumber'] as int;
 
     var url = _urlForScriptId(scriptId);
-    return _locations.locationForJs(url, lineNumber + 1);
+    return _locations.locationForJs(url, line + 1, column + 1);
```
Comment (nshahan) [inline dwds/lib/src/debugging/debugger.dart:335]: Might be nice to copy the comment here that I see below about the values being zero based in chrome devtools.

**Review by nshahan — APPROVED** [review]
Thanks Anna! LGTM with a few comments.

File: dwds/lib/src/debugging/debugger.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart:733]: thanks!

File: dwds/test/build_daemon_breakpoint_test.dart (hunk 2 above)
Comment (annagrin) [reply dwds/test/build_daemon_breakpoint_test.dart:91]: The column returned does not currently match the column we tried to use when setting a breakpoint, the tools tries to match the first column greater or equal then the one we asked for that we can set the breakpoint at. I will add that expectation, thanks!

File: dwds/test/frontend_server_breakpoint_test.dart (hunk 3 above)
Comment (annagrin) [reply dwds/test/frontend_server_breakpoint_test.dart:122]: Sure!

File: dwds/lib/src/debugging/debugger.dart (hunk 4 above)
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart:335]: will do.

Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart:335]: Actually, the original JS locations are 0-based but we convert them to 1-based in `JsLocation.fromZeroBased`. I removed the original adjustment so we don't need to do it here or in other places either.

File: dwds/lib/src/debugging/debugger.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart]: Done!

File: dwds/test/build_daemon_breakpoint_test.dart (hunk 2 above)
Comment (annagrin) [reply dwds/test/build_daemon_breakpoint_test.dart]: done!

File: dwds/test/frontend_server_breakpoint_test.dart (hunk 3 above)
Comment (annagrin) [reply dwds/test/frontend_server_breakpoint_test.dart]: done!

**Review by elliette — APPROVED** [review]

**Pull request merged**